### PR TITLE
feat: Adds Testing page

### PR DESCRIPTION
**Pull request by dhudec**
## Description

- Documents testing recommendations
- Documents test card numbers

## Testing required outside of automated testing?

- [ ] Not Applicable

### Screenshots (if appropriate):

- [ ] Not Applicable

## Rollback / Rollforward Procedure

- [ ] Roll Forward
- [ ] Roll Back

## Reviewer Checklist

- [ ] Description of Change
- [ ] Description of outside testing if applicable.
- [ ] Description of Roll Forward / Backward Procedure
- [ ] Documentation updated for Change

### CODE REVIEW

File: docs/api/testing.mdx [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+
+# Testing
+
+## Recommended Tenant Setup
+
+When integration testing or running your systems within a deployed environment, we recommend that you use a
+dedicated Basis Theory [tenant](/docs/concepts/access-controls#what-are-tenants) for each
+test environment that is separate from your production tenant. For example, if you maintain a separate Development,
+QA, Staging, and Production environment for your systems, we recommend that you mirror this setup by creating 4 separate tenants in Basis Theory.
+This can help you to isolate test data from production data and allow you to more securely lock down access to production API keys.
```
Comment (jleon15) [inline docs/api/testing.mdx:13]: is this `to` needed?

File: docs/api/testing.mdx [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+
+## Testing Reactors
```
Comment (jleon15) [inline docs/api/testing.mdx:15]: what about proxies?

Comment (dhudec) [reply docs/api/testing.mdx:15]: I don't have time to write about that too. All of this was extra and unplaned as it is.

File: docs/api/testing.mdx (hunk 1 above)
Comment (dhudec) [reply docs/api/testing.mdx:13]: either way is correct

**Review by djejaquino — APPROVED** [review]

**Pull request merged**